File: 70. Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        result = [1,2]
        for i in range(2,n):
            result.append(result[i-1]+result[i-2])
        return result[n-1]

class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1 or n == 2:
            return n
        a, b = 1, 2
        for _ in range(2,n):
            c = a + b
            a = b
            b = c
        return c

File: 70. Climbing Stairs.java
/**
 * dp solution 
 * Time O(n)
 * Space O(1)
 */
class Solution {
    public int climbStairs(int n) {
        if (n==1 || n==2) {
            return n;
        }
        int n_2 = 1;
        int n_1 = 1;
        int count = 1;
        while (count < n) {
            int temp = n_2 + n_1;
            n_2 = n_1;
            n_1 = temp;
            count++;
        }
        return n_1;
    }
}